esp_jpeg/include/jpeg_decoder.h

```diff
@@ -48,6 +48,12 @@ typedef struct esp_jpeg_image_cfg_s {
         uint8_t swap_color_bytes: 1; /*!< Swap first and last color bytes */
     } flags;
 
+    struct {
+        size_t working_buffer_size; /*!< Size of working memory for the decoder. Set to 0 for default value.
+                                         Default size is 3.1kB or 65kB if JD_FASTDECODE == 2
+                                         Tjpgd does not use dynamic allocation, se we pass this buffer to Tjpgd that uses it as scratchpad */
```

Comment:
```suggestion
                                         Tjpgd does not use dynamic allocation, so we pass this buffer to Tjpgd that uses it as scratchpad */
```

Reply:
I checked memory allocations inside Tjpgd and the required working buffer size is sum of the following:
```
JD_SZBUF bytes for input buffer
256 bytes for de-quantization table
16 bytes for bit distribution table
5 x number_of_code_words for decoding
up to 256 bytes for decoding scratchpad
(number_of_blocks (typical 8x8 or 16x8) + 2) x 64 bytes
```
So it only depends on the JPG image. There is no way to predict the required working buffer size. Image size independent.

Moreover, a typical user has no clue what these values mean (I didn't until few weeks ago :D) 

Taking this into consideration, the best advice is 'If your decoder returns `JDR_MEM1` or `JDR_MEM2` increase the working buffer size 🤷

Reply:
Can we split the process of decoding the image into more steps

1. Load the image header
2. Application calls `get_required_buffer_size(image_handle)` and we calculate the size based on the information above. Application allocates the buffer of the required size
2. Decode the rest of the image with the application-provided buffer.

Similar to how this is done in libpng:
https://github.com/espressif/idf-extra-components/blob/9634f8d6c04c53d82a2036ed6800d357d89c0d5c/libpng/test_apps/main/test_libpng.c#L30-L42

Reply:
Yes, that would be the best. It requires more work than this simple hack, though... I'll give it more time

Reply:
If the current PR already helps you achieve your goal, I don't mind merging it as is. We can leave this for later.